main/streams/plain_wrapper.c

```diff
@@ -840,14 +840,15 @@ static int php_stdiop_set_option(php_stream *stream, int option, int value, void
 							gran = info.dwAllocationGranularity;
 							loffs = ((DWORD)range->offset / gran) * gran;
 							delta = (DWORD)range->offset - loffs;
+							hoffs = range->offset >> 32;
```

Comment:
I think this code is correct as written, but it's not really obvious and relies on gran being a power of two (which should be a given, of course). I think it would be more obvious to do something like this:
```suggestion
							size_t rounded_offset = (range->offset / gran) * gran;
							delta = range->offset - rounded_offset;
							loffs = (DWORD)rounded_offset;
							hoffs = (DWORD)(rounded_offset >> 32);
```

Reply:
ACK, makes sense. Maybe also assert that gran is a power of 2 (`ZEND_ASSERT(gran != 0 && gran & (gran - 1) == 0)`), and make use of that property when rounding (`rounded_offset = range->offset & ~((size_t) gran - 1)`)?

Reply:
Given what the rest of the code does, I don't think it's super important to optimize away the division here.

Reply:
Right.